### ISSUE TITLE
production release plan

### ISSUE THREAD

**Issue by metadave** [mozmeao/infra#410]
* Prerequisites 
   * [x] Maintenance Mode 100% traffic test ([milestone 12](https://github.com/mozmeao/infra/milestone/12))
   * [x] Manual test of staging in Read / Write mode ([milestone 13](https://github.com/mozmeao/infra/milestone/13))
   * [x] SLA / On Call agreement (@bensternthal to lead) (issue #560)
   * [x] Date set, announced on developer.mozilla.org via Soapbox, webops informed

* Preparation
   - [x] Merge freeze on [kuma](https://github.com/mozilla/kuma), [kumascript](https://github.com/mdn/kumascript)
   - [x] Production Portland k8s images set to correct tag 
   - [x] Production Portland k8s in maintenance mode configuration
   - [x] merge Jenkins pipeline branch for kuma/kumascript
   - [x] Rebase [maintenance_mode](https://github.com/mozilla/kuma/tree/maintenance_mode) branch on current master
* SCL3 in Maintenance Mode
  - [x] Inform #moc, #webops, #mdn IRC channels
  - [x] Deploy [maintenance_mode](https://github.com/mozilla/kuma/tree/maintenance_mode) branch to SCL3
  - [x] Shut down ``celerybeat``, ``celerycam`` on developer-celery1
* Transfer from SCL3 to AWS
  - [x] Break replication on MySQL replica
  - [x] Create MySQL backup (compressed, encrypted)
  - [x] Transfer to S3
  - [ ] Load into production RDS instance
  - [x] Synchronize Files
    - [x] push from SCL3
    - [x] pull to AWS
* Send Traffic to AWS
  - [ ] Switch from 100% SCL3 policy to 100% Portland k8s policy for developer.mozilla.org
  - [ ] Update Attachments / Samples domain (mdn.mozillademos.org) from SCL3 to Portland k8s
  - [ ] Monitor in New Relic, Sentry, AWS, and Papertrail
  - [ ] Inform #moc, #webops, #mdn IRC channels
* Cleanup
  - [ ] Remove (large) backup files from AWS prod
  - [ ] disable SCL3 sync crontask

**Comment by jwhitlock** [comment]
On Aug 11, @metadave said:

> Lets work on the [maintenance mode release plan](https://github.com/mozmeao/infra/issues/409) first, and then update this issue accordingly.

We're done w/ maintenance mode testing. I propose the task description should be the latest version of the production checklist. We propose changes by editing it, and maybe discussing with a comment. That way, we have a working checklist on go-live day.

**Comment by metadave** [comment]
I've added some updates to the list above.

---

> Break replication on MySQL replica

Are there any docs on how to disable replication?

> Update CDN (developer.cdn.mozilla.net) source from SCL3 to Portland k8s

The developer.cdn.mozilla.net origin is set to `developer.mozilla.org`. I confirmed via the current Cloudformation json that creates developer.cdn.mozilla.net (I can provide a link privately).

**Comment by metadave** [comment]
docs on disabling replication: https://dev.mysql.com/doc/refman/5.7/en/stop-slave.html

**Comment by metadave** [comment]
I'm working with the data team to get the appropriate permissions.

**Comment by metadave** [comment]
we may need to merge Jenkins branches once Jenkins pipeline work is completed: https://github.com/mozmeao/infra/issues/394

**Comment by metadave** [comment]
Chris Valaas and Matt Pressman will be joining us on Tuesday to stop replication

**Comment by escattone** [comment]
https://github.com/mozmeao/infra/pull/565 provides the two configurations we'll use on "go-live" day.

**Comment by metadave** [comment]
MDN is now running in AWS.

🎉🎉🎉🎉

**Comment by jwhitlock** [comment]
The AWS transition had one hiccup. This was the first time we had served production-level traffic for mdn.mozillademos.org, and AWS was not able to handle it. Eventually, we increased the web capacity from 6 pods with 4 workers each to 16 pods with 8 workers each, and were able to handle the file serving traffic reliably.  We were able to switch datacenters within the announced maintenance window.

There are additional high-priority issues, but we're now in the [MDN Post-Migration](https://github.com/mozmeao/infra/projects/5) phase. 😎 

@metadave - are the cleanup tasks done, or do they need to be tracked in new issues?
